lib/NQP/Config.pm

```diff
@@ -247,16 +247,23 @@ sub make_cmd {
         my $has_gcc   = 0 == system('gcc --version >NUL 2>&1');
         if ($has_cl) {
             if ( $cl_report =~
-                /Microsoft\s.*\sCompiler\s+Version\s+(\d+(?:\.\d+)+)/i )
+                /Microsoft\s.*\sCompiler\s+Version\s+(\d+(?:\.\d+)+)\s+for\s+(x\d+)/i )
             {
                 my $actual_version = $1;
+		my $cl_arch        = $2;
                 my $expect_version = "19.0";
+		
                 if ( version->parse($actual_version) < $expect_version ) {
                     $self->sorry( "Expected Microsoft Compiler version "
                           . $expect_version
                           . "+, but got "
                           . $actual_version );
                 }
+                if ( $cl_arch ne "x64" ) {
+                    $self->sorry( "Rakudo compiles currently only on "
+                          . "x64 architectures, but we are on "
```

Comment:
I've built rakudo on arm64 Windows before. If I remember correctly it requires using c11-atomics (which is how I ended up writing [this PR](https://github.com/MoarVM/MoarVM/pull/1777) and an [updated version of dyncall](https://github.com/MoarVM/MoarVM/pull/1777)

Reply:
Also if I understand correctly the issue you linked to in MoarVM is related to building libuv, yet one can pass an option to use an external copy of libuv. We shouldn't block the user from installing on architectures due to a flaw in how we build a 3rd party dependency.

Reply:
> I've built rakudo on arm64 Windows before. If I remember correctly it requires using c11-atomics (which is how I ended up writing [this PR](https://github.com/MoarVM/MoarVM/pull/1777)) and an [updated version of dyncall](https://irclogs.raku.org/moarvm/2023-11-11.html#03:11)

Wow, cool, thanks for this info!!!

I have tried to build Rakudo & friends on my arm-based Lenovo Windows10 box, which emulates Windows x86 / 32bit, some months ago. This definitely didn't build / compile.

Have adopted the regex'es now to match [\D+64$] ... this should work for arm64 then as well...

Reply:
> Also if I understand correctly the issue you linked to in MoarVM is related to building libuv, yet one can pass an option to use an external copy of libuv. We shouldn't block the user from installing on architectures due to a flaw in how we build a 3rd party dependency, particularly if that dependency can be provided in another way.

Well, that's just an example... I've tried to build Rakudo on some of my older 32bit Windows boxes (up to Windows10) in the past and never succeeded.. at that time, I was told, Rakudo doesn't compile on Win 32bit "per purpose"... and if this is the case, let's implement a check in parallel to the CL.EXE version...